src/XRP/issued-currency-client.ts

```diff
@@ -1024,4 +1033,107 @@ export default class IssuedCurrencyClient {
+   *               Use a string to specify XRP, or an IssuedCurrency object to specify an issued currency.
+   * @param offerSequence (Optional) An offer to delete first, specified by the sequence number of a previous OfferCreate transaction.
+   *               If specified, cancel any offer object in the ledger that was created by that transaction.
+   *               It is not considered an error if the offer specified does not exist.
+   * @param expiration (Optional) Time after which the offer is no longer active, in seconds since the Ripple Epoch.
+   *               (See https://xrpl.org/basic-data-types.html#specifying-time)
+   */
+  public async createOffer(
+    sender: Wallet,
+    takerGetsAmount: string | IssuedCurrency,
+    takerPaysAmount: string | IssuedCurrency,
+    offerSequence?: number,
+    expiration?: number,
+  ): Promise<TransactionResult> {
+    // construct TakerGets
+    const takerGetsCurrencyAmount = new CurrencyAmount()
+    if (typeof takerGetsAmount == 'string') {
+      const takerGetsXrpDropsAmount = new XRPDropsAmount()
+      takerGetsXrpDropsAmount.setDrops(takerGetsAmount)
+      takerGetsCurrencyAmount.setXrpAmount(takerGetsXrpDropsAmount)
+    } else {
+      const takerGetsCurrency = new Currency()
+      takerGetsCurrency.setName(takerGetsAmount.currency)
+
+      const takerGetsAccountAddress = new AccountAddress()
+      takerGetsAccountAddress.setAddress(takerGetsAmount.issuer)
+
+      const takerGetsIssuedCurrencyAmount = new IssuedCurrencyAmount()
+      takerGetsIssuedCurrencyAmount.setIssuer(takerGetsAccountAddress)
+      takerGetsIssuedCurrencyAmount.setCurrency(takerGetsCurrency)
+      takerGetsIssuedCurrencyAmount.setValue(takerGetsAmount.value)
+
+      takerGetsCurrencyAmount.setIssuedCurrencyAmount(
+        takerGetsIssuedCurrencyAmount,
+      )
+    }
+    const takerGets = new TakerGets()
+    takerGets.setValue(takerGetsCurrencyAmount)
+
+    // construct TakerPays
+    const takerPaysCurrencyAmount = new CurrencyAmount()
+    if (typeof takerPaysAmount == 'string') {
+      const takerPaysXrpDropsAmount = new XRPDropsAmount()
+      takerPaysXrpDropsAmount.setDrops(takerPaysAmount)
+      takerPaysCurrencyAmount.setXrpAmount(takerPaysXrpDropsAmount)
+    } else {
+      const takerPaysCurrency = new Currency()
+      takerPaysCurrency.setName(takerPaysAmount.currency)
+
+      const takerPaysAccountAddress = new AccountAddress()
+      takerPaysAccountAddress.setAddress(takerPaysAmount.issuer)
+
+      const takerPaysIssuedCurrencyAmount = new IssuedCurrencyAmount()
+      takerPaysIssuedCurrencyAmount.setIssuer(takerPaysAccountAddress)
+      takerPaysIssuedCurrencyAmount.setCurrency(takerPaysCurrency)
+      takerPaysIssuedCurrencyAmount.setValue(takerPaysAmount.value)
+
+      takerPaysCurrencyAmount.setIssuedCurrencyAmount(
+        takerPaysIssuedCurrencyAmount,
+      )
+    }
```

Comment:
Since this logic is identical for takerGets and takerPays, you could put it in a helper function

Reply:
good idea, done